### PR TITLE
CC-472 Add reply prop and rename more_info prop to description in Listing Teaser

### PR DESCRIPTION
**Pull request by mikemai2awesome**
## Jira

https://pegadigitalit.atlassian.net/browse/CC-472

## Summary

Adds a `reply` prop, and rename `more_info` to `description` in the Listing Teaser component

## Details

1. Added `reply` prop to showcase a relevant reply.
2. Updated CSS to style the reply content.
3. Added docs to demonstrate the `reply` prop.
4. `more_info` has been renamed to `description`, its type has been changed to `any`.
5. `warning` prop's type has been changed to `any`.

## How to test

Run the branch locally and read the Listing Teaser props, especially the one with reply prop and Collaboration Center use case. Make sure the prop works as expected.

### CODE REVIEW

File: packages/components/bolt-listing-teaser/src/listing-teaser.twig [hunk 1]
```diff
@@ -82,6 +82,26 @@
           } only %}
         </div>
       {% endif %}
+      {% if this.data.reply %}
```
Comment (adamszalapski) [inline packages/components/bolt-listing-teaser/src/listing-teaser.twig:85]: We shouldn't use `this.data` for checking content different than `string`. @danielamorse, @remydenton please correct me if I'm wrong. In this case, we should use `{% if reply %}`

File: packages/components/bolt-listing-teaser/src/listing-teaser.twig [hunk 2]
```diff
@@ -82,6 +82,26 @@
...
+          {% set _icon_chevron_right %}{% spaceless %}
+            {% include '@bolt-components-icon/icon.twig' with {
+              name: 'chevron-right',
+            } only %}
+          {% endspaceless %}{% endset %}
+          {% set _reply_link %}
+            <span class="e-bolt-text-link e-bolt-text-link--reversed-underline">{{ 'Go to reply'|t }}<span class="e-bolt-text-link__icon-after">{{ _icon_chevron_right }}</span></span>
+          {%  endset %}
+          {% include '@bolt-components-stack/stack.twig' with {
+            content: this.data.reply,
```
Comment (adamszalapski) [inline packages/components/bolt-listing-teaser/src/listing-teaser.twig:96]: The same issue here we should pass `content: reply`

File: packages/components/bolt-listing-teaser/src/listing-teaser.twig [hunk 3]
```diff
@@ -82,6 +82,26 @@
...
         </div>
       {% endif %}
+      {% if this.data.reply %}
+        <div class="c-bolt-listing-teaser__flag-content__item c-bolt-listing-teaser__flag-content__item--reply">
+          {% set _icon_chevron_right %}{% spaceless %}
+            {% include '@bolt-components-icon/icon.twig' with {
+              name: 'chevron-right',
+            } only %}
+          {% endspaceless %}{% endset %}
+          {% set _reply_link %}
```
Comment (adamszalapski) [inline packages/components/bolt-listing-teaser/src/listing-teaser.twig:92]: I noticed that right now `Go to reply` will point to the same place as a whole row (whole row clickability) but what I heard was that header should link to the post where the answer was added and `Go to replay` should point to that reply directly. I don't know if you know it @mikemai2awesome or maybe you agreed that we will change the existing approach.

**Review by adamszalapski — CHANGES_REQUESTED** [review]
@mikemai2awesome I add few comments, please check.

File: packages/components/bolt-listing-teaser/src/listing-teaser.twig (hunk 3 above)
Comment (adamszalapski) [reply packages/components/bolt-listing-teaser/src/listing-teaser.twig:92]: Maybe we should add `reply_link` prop?

Comment (mikemai2awesome) [reply packages/components/bolt-listing-teaser/src/listing-teaser.twig:92]: thanks for the heads up. let me make some changes

File: packages/components/bolt-listing-teaser/src/listing-teaser.twig (hunk 1 above)
Comment (mikemai2awesome) [reply packages/components/bolt-listing-teaser/src/listing-teaser.twig]: all set

File: packages/components/bolt-listing-teaser/src/listing-teaser.twig (hunk 2 above)
Comment (mikemai2awesome) [reply packages/components/bolt-listing-teaser/src/listing-teaser.twig]: all set

File: packages/components/bolt-listing-teaser/src/listing-teaser.twig (hunk 3 above)
Comment (mikemai2awesome) [reply packages/components/bolt-listing-teaser/src/listing-teaser.twig]: prop is updated

**Review by adamszalapski — APPROVED** [review]
Looks good, great job @mikemai2awesome

**Review by colbytcook — APPROVED** [review]

**Review by remydenton — APPROVED** [review]
Looks great, nice work.

**Review by adamszalapski — APPROVED** [review]
Everything ok, nice work @mikemai2awesome

**Review by remydenton — APPROVED** [review]

**Pull request merged**